types/datum_test.go

```diff
@@ -151,6 +151,30 @@ func TestToInt64(t *testing.T) {
 	testDatumToInt64(t, v, int64(3))
 }
 
+func TestToUint32(t *testing.T) {
+	t.Parallel()
+
+	sc := new(stmtctx.StatementContext)
+	sc.IgnoreTruncate = true
+
+	// test overflow
+	var datum = NewUintDatum(5000000000)
```

Comment:
Please make this test more scalable and add more test cases for it.

Reply:
Done. PTAL @wjhuang2016